src/codemodder/codemods/imported_call_modifier.py

```diff
@@ -0,0 +1,90 @@
+import abc
+from typing import Sequence
+
+import libcst as cst
+from libcst import matchers
+from libcst.codemod import CodemodContext, VisitorBasedCodemodCommand
+from libcst.metadata import PositionProvider
+
+from codemodder.change import Change
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+from codemodder.file_context import FileContext
+
+
+class ImportedCallModifier(
+    VisitorBasedCodemodCommand, NameResolutionMixin, metaclass=abc.ABCMeta
+):
+    METADATA_DEPENDENCIES = (PositionProvider,)
+
+    def __init__(
+        self,
+        codemod_context: CodemodContext,
+        file_context: FileContext,
+        matching_functions: dict[str, str],
+        change_description: str,
+    ):
+        super().__init__(codemod_context)
+        self.line_exclude = file_context.line_exclude
+        self.line_include = file_context.line_include
+        self.matching_functions = matching_functions
+        self._matching_function_names = list(matching_functions.keys())
+        self.change_description = change_description
+        self.changes_in_file: list[Change] = []
+
+    def updated_args(self, original_args: Sequence[cst.Arg]):
+        return original_args
+
+    @abc.abstractmethod
+    def update_attribute(self, true_name, original_node, updated_node, new_args):
+        """Callback to modify tree when the detected call is of the form a.call()"""
+
+    @abc.abstractmethod
+    def update_simple_name(self, true_name, original_node, updated_node, new_args):
+        """Callback to modify tree when the detected call is of the form call()"""
+
+    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
+        pos_to_match = self.node_position(original_node)
+        line_number = pos_to_match.start.line
+        if self.filter_by_path_includes_or_excludes(pos_to_match):
+            true_name = self.find_base_name(original_node.func)
+            if (
+                self._is_direct_call_from_imported_module(original_node)
+                and true_name in self._matching_function_names
```

Comment:
Changing it to a list here will degrade the `in` operation from O(1) to O(n) (average case). Either use the original `matching_functions` attribute or at least make `_matching_function_names` a set.

Also, I'm not sure I understood why you've felt the need to create `_matching_function_names` in the first place.